packages/cli/src/utils/registry/index.ts

```diff
@@ -37,44 +41,74 @@ export async function fetchRegistry() {
  * @returns An array of component objects matching the provided component names.
  */
 export async function getComponentInfo(componentName: string[] | undefined) {
-  const componentRegistry = registryIndexSchema.parse(await fetchRegistry());
+  // const componentRegistry = registryIndexSchema.parse(await fetchRegistry());
+  const componentRegistry = registryIndexSchema.parse(ui);
+
+  const addedComponents = new Set<string>(componentName); // Set to store added component names
 
   if (!componentName) {
-    console.log("no components selected");
+    console.log("No components selected");
     process.exit(1);
   }
+
   const tree: z.infer<typeof registryIndexSchema> = [];
+
   for (const name of componentName) {
     const component = componentRegistry.find(
       (component: { name: string }) => component.name === name
     );
+
     if (!component) {
-      // @FIXME: If we're going to exit, we should do so with a non-zero exit code
-      // i.e. process.exit(1)?
       console.log(`Component ${name} not found in registry`);
       continue;
-    } else {
-      // only push the component if it's found
-      tree.push(component);
+    }
+
+    tree.push(component);
+
+    if (component.registryDependencies) {
+      const dependencies = await getComponentInfo(
+        component.registryDependencies
+      );
+      for (const dependency of dependencies) {
+        if (addedComponents.has(dependency.name)) {
+          continue;
+        }
+        tree.push(...dependencies);
+      }
     }
   }
+
   return tree;
 }
 
+/**
+ * gets paths from registry components
+ *
+ * @export
+ * @param {Registry} components
+ * @return {*}
+ */
 export function getPaths(components: Registry) {
   const pathArray: string[] = components.map((obj) => obj.files).flat();
   return pathArray;
 }
-
+/**
+ *
+ * fetches raw files from github using the paths provided from registry
+ * @export
+ * @param {string[]} paths
+ * @return {*}  {Promise<{ filenames: string[], contents: string[] }>}
+ */
 export async function fetchFileContentFromGithub(
   paths: string[]
-): Promise<string[]> {
+): Promise<{ filenames: string[]; contents: string[] }> {
   try {
+    const filenames: string[] = [];
     const contents: string[] = [];
 
     for (const path of paths) {
       const rawUrl = `${GithubUrl}/packages/site/src/registry/alpha/${path}`;
-
+      const filename = path.substring(path.indexOf("/", path.indexOf("/") + 1));
```

Comment:
What are we doing here, looks like we're looking to see if 2 `/` exist in the path?

Let's run this by AI and see if there is a more readable way of doing the same thing.

Reply:
made it a little bit readable